### PR TITLE
chore(cketh): Proposal to upgrade all ledger suites

### PR DESCRIPTION
**Pull request by gregorydemay**
Proposal to let the ckETH ledger suite be managed by the ledger sutie orchestrator and upgrade all managed ledger suites to the latest version.

### CODE REVIEW

File: rs/ethereum/ledger-suite-orchestrator/tests/tests.rs [hunk 1]
```diff
@@ -1114,24 +1114,22 @@ mod upgrade {
...
-                UpgradeArg {
-                    git_commit_hash: Some(GIT_COMMIT_HASH_UPGRADE.to_string()),
-                    ledger_compressed_wasm_hash: Some(embedded_ledger_wasm_hash.to_string()),
-                    index_compressed_wasm_hash: Some(embedded_index_wasm_hash.to_string()),
-                    archive_compressed_wasm_hash: Some(embedded_archive_wasm_hash.to_string()),
-                    cycles_management: None,
-                    manage_ledger_suites: None,
-                },
-            );
+        let orchestrator = orchestrator.upgrade_ledger_suite_orchestrator(
```
Comment (gregorydemay) [inline rs/ethereum/ledger-suite-orchestrator/tests/tests.rs:1117]: little refactoring to ensure that it works with a single proposal (single upgrade)

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_09_20.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+2. Upgrade all ledger suites managed by the orchestrator to the latest version to start the migration towards stable memory.
+
+## Upgrade args
+
+```
+git fetch
+git checkout d4ee25b0865e89d3eaac13a60f0016d5e3296b31
+cd rs/ethereum/ledger-suite-orchestrator
+didc encode -d ledger_suite_orchestrator.did -t '(OrchestratorArg)' '(variant { UpgradeArg = record {git_commit_hash = opt "d4ee25b0865e89d3eaac13a60f0016d5e3296b31"; ledger_compressed_wasm_hash = opt "e6072806ae22868ee09c07923d093b1b0b687dba540d22cfc1e1a5392bfcca46"; index_compressed_wasm_hash = opt "de250f08dc7e699144b73514f55fbbb3a3f8cd97abf0f7ae31d9fb7494f55234"; archive_compressed_wasm_hash = opt "e9c7cad647ede2ea2942572f337bd27d0839dd06c5e2c7f03591226acb10a9fb"; manage_ledger_suites = opt vec { record { token_symbol = "ckETH"; ledger = record { canister_id = principal "ss2fx-dyaaa-aaaar-qacoq-cai"; installed_wasm_hash = "8457289d3b3179aa83977ea21bfa2fc85e402e1f64101ecb56a4b963ed33a1e6"}; index = record { canister_id = principal "s3zol-vqaaa-aaaar-qacpa-cai"; installed_wasm_hash = "eb3096906bf9a43996d2ca9ca9bfec333a402612f132876c8ed1b01b9844112a"}; archives = opt vec { principal "xob7s-iqaaa-aaaar-qacra-cai" }}}}})'
```
Comment (THLO) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_09_20.md:24]: 4449444c136b03ba9edbce040186dac7c6090cc3cab9d20a106c06b786991c02b59798d00208e898cad2040be9c2d6a7090ba1eedde50a0b98c6c2cf0b0b6e036d046c049efeb9a40371a9cbadc3090592b6d2f00b05f1ecb9eb0e066c02b3c4b1f2046884f1dcc60a716e076d686e096c04e7aba2140acdd1f4e2040ae4d8cce80b0a94e2c9d00c0a6e7d6e716c03b59798d0020dffb08aab0807cb83d3c90f0f6e0e6c04e7aba2147dcdd1f4e2047de4d8cce80b7d94e2c9d00c7d6e686c0292bec7bd0311fdb5c9a907126c0299e7a4cf087db4e3ade809716c05c295a993017b9efeb9a40371f2c794ae037db1a38bf40d7191c9aafe0d71010000010105636b455448010a000000000230009d01014038343537323839643362333137396161383339373765613231626661326663383565343032653166363431303165636235366134623936336564333361316536010a000000000230009e010140656233303936393036626639613433393936643263613963613962666563333333613430323631326631333238373663386564316230316239383434313132610101010a00000000023000a2010100014065396337636164363437656465326561323934323537326633333762643237643038333964643036633565326337663033353931323236616362313061396662012864346565323562303836356538396433656161633133613630663030313664356533323936623331014065363037323830366165323238363865653039633037393233643039336231623062363837646261353430643232636663316531613533393262666363613436014064653235306630386463376536393931343462373335313466353566626262336133663863643937616266306637616533316439666237343934663535323334

Comment (gregorydemay) [reply rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_09_20.md:24]: confirmed!

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_09_20.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+## Wasm Verification
+
+Verify that the hash of the gzipped WASM matches the proposed hash.
+
+```
+git fetch
+git checkout d4ee25b0865e89d3eaac13a60f0016d5e3296b31
+./ci/container/build-ic.sh -c
+sha256sum ./artifacts/canisters/ic-ledger-suite-orchestrator-canister.wasm.gz
```
Comment (THLO) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_09_20.md:85]: ``£
$ sha256sum ./artifacts/canisters/ic-ledger-suite-orchestrator-canister.wasm.gz
61a422257859c91ba01bb082c6daaae986a5ca7afcd9d0293dc9bc199e31c8c4  ./artifacts/canisters/ic-ledger-suite-orchestrator-canister.wasm.gz
```

Confirmed!

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_09_20.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+Verify that the hash of the gzipped WASM for the ledger, index and archive match the proposed hash.
+
+```
+git fetch
+git checkout d4ee25b0865e89d3eaac13a60f0016d5e3296b31
+./ci/container/build-ic.sh -c
+sha256sum ./artifacts/canisters/ic-icrc1-ledger-u256.wasm.gz
+sha256sum ./artifacts/canisters/ic-icrc1-index-ng-u256.wasm.gz
+sha256sum ./artifacts/canisters/ic-icrc1-archive-u256.wasm.gz
```
Comment (THLO) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_09_20.md:96]:
```
$ sha256sum ./artifacts/canisters/ic-icrc1-ledger-u256.wasm.gz
e6072806ae22868ee09c07923d093b1b0b687dba540d22cfc1e1a5392bfcca46  ./artifacts/canisters/ic-icrc1-ledger-u256.wasm.gz
$ sha256sum ./artifacts/canisters/ic-icrc1-index-ng-u256.wasm.gz
de250f08dc7e699144b73514f55fbbb3a3f8cd97abf0f7ae31d9fb7494f55234  ./artifacts/canisters/ic-icrc1-index-ng-u256.wasm.gz
$ sha256sum ./artifacts/canisters/ic-icrc1-archive-u256.wasm.gz
e9c7cad647ede2ea2942572f337bd27d0839dd06c5e2c7f03591226acb10a9fb  ./artifacts/canisters/ic-icrc1-archive-u256.wasm.gz
```

Confirmed!

**Review by THLO — APPROVED** [review]

**Pull request merged**